src/App.tsx

```diff
@@ -1,26 +1,296 @@
+  const [isEdited, setIsEdited] = useState<number | null>(null);
+  // const [todos, setTodos] = useState<Todo[] | []>([]);
+  const [editedTitle, setEditedTitle] = useState('');
+  const [newTitle, setNewTitle] = useState('');
+  const [errMessage, setErrMessage] = useState('');
+  const [stat, setStat] = useState(Status.all);
+  const [visibleErr, setVisibleErr] = useState(false);
+  const [tempTodo, setTempTodo] = useState<Todo[] | null>(null);
+  const [todos, setTodos] = useState<Todo[] | []>([]);
+
+  const filteredTodos = (tod: Todo[], type: Status) => {
+    switch (type) {
+      case Status.active:
+        return tod.filter(todo => !todo.completed);
+      case Status.completed:
+        return tod.filter(todo => todo.completed);
+
+      default:
+        return tod;
+    }
+  };
+
+  const visibleTodos = tempTodo
+    ? filteredTodos(tempTodo, stat)
+    : filteredTodos(todos, stat);
+
+  const editSelectedInput = useRef<HTMLInputElement>(null);
+  const isAnyCompleted = todos.some(todo => todo.completed);
+
+  const resetErr = () =>
+    setTimeout(() => {
+      setVisibleErr(false);
+      setErrMessage('');
+    }, 3000);
+
+  useEffect(() => {
+    const loadTodos = async () => {
+      getTodos()
+        .then(setTodos)
+        .catch(() => {
+          setVisibleErr(true);
+          setErrMessage('Unable to load todos');
+          resetErr();
+        });
+    };
+
+    loadTodos();
+  }, []);
+
+  const updateTodo = async (updatedTodo: Todo, option: keyof Todo) => {
+    try {
+      setIsLoading(updatedTodo.id);
+      const todoToUpdate = todos.find(tod => tod.id === updatedTodo.id);
+
+      let newTodo: Todo = {
+        id: 0,
+        userId: 0,
+        title: '',
+        completed: false,
+      };
+
+      if (option === 'completed' && todoToUpdate) {
+        newTodo = { ...todoToUpdate, completed: !updatedTodo.completed };
+      }
+
+      if (option === 'title' && todoToUpdate) {
+        newTodo = { ...todoToUpdate, title: editedTitle };
+      }
+
+      await patchTodo(newTodo);
+
+      await getTodos().then(setTodos);
+
+      // setTodos(prevTodos => {
+      //   return prevTodos.map(todo => {
+      //     if (todo.id === updatedTodo.id) {
+      //       return { ...todo, ...newTodo };
+      //     }
+
+      //     return todo;
+      //   });
+      // });
+    } catch {
+      setVisibleErr(true);
+      setErrMessage('Unable to update a todo');
+      resetErr();
+    } finally {
+      setIsLoading(null);
+    }
+  };
+
+  const addTodo = async () => {
+    const trimedTitle = newTitle.trim();
+
+    if (trimedTitle === '') {
+      setVisibleErr(true);
+      setErrMessage('Title should not be empty');
+      resetErr();
+
+      return;
+    }
+
+    if (trimedTitle !== '') {
+      const newTodo: Todo = {
+        id: 0,
+        userId: 472,
+        title: trimedTitle,
+        completed: false,
+      };
+
+      const temp = {
```

Comment:
Why do you need newTodo and temp? 
Looks like they are pretty much the same items